core/src/main/scala/caliban/parsing/parsers/Parsers.scala

```diff
@@ -348,7 +348,7 @@ object Parsers extends SelectionParsers {
     schemaExtension | typeExtension
 
   def definition(implicit ev: P[Any]): P[Definition] =
-    executableDefinition | typeSystemDefinition | typeSystemExtension
+    typeSystemDefinition | typeSystemExtension
```

Comment:
It doesn't match the spec, how about changing `document` instead?

Reply:
I reverted the change. _tbh_ it only makes a difference for invalid queries really

Reply:
But how about changing `document` to `(Start ~ definition.rep ~ End)`? I agree with your change, just wanted to apply it somewhere else 😄

Reply:
`document` is currently this:

```scala
((Start ~ executableDefinition.rep ~ End) | (Start ~ definition.rep ~ End)).map(seq => ParsedDocument(seq.toList))
```

The reason for having `Start ~ executableDefinition.rep ~ End` separately is so that when when we reach the end of the file after `executableDefinition` has successfully parsed (which is the most common thing we parse), we shortcut to the end and don't try `typeSystemDefinition` prior to exiting. I'm not sure if it's even possible, but in the case that there is a typeSystem definition in the query then we'll fallback to the full parser.

I actually realised that the "fix" I did earlier wouldn't work for cases that there is a mix of them; although I don't know if that's a valid query 🤔. If yes I should add a test for it

Reply:
Ah, didn't know we were trying to be clever 🤣